src/ACEtk/block/AngularDistributionData.hpp

```diff
@@ -169,19 +169,25 @@ class AngularDistributionData : protected Base {
   Distribution angularDistributionData( std::size_t index ) const {
 
     const auto type = this->angularDistributionType( index );
-    const auto locator = this->relativeAngularDistributionLocator( index );
     const double incident = this->incidentEnergy( index );
     if ( type == AngularDistributionType::Isotropic ) {
 
         return IsotropicAngularDistribution( incident );
     }
     else {
 
+      const auto locator = this->relativeAngularDistributionLocator( index );
       const auto left = std::next( this->begin(), locator - 1 );
-      const auto right = index == this->numberIncidentEnergies()
-          ? this->end()
-          : std::next( this->begin(),
-                       ( this->relativeAngularDistributionLocator( index + 1 ) - 1 ) );
+      auto right = this->end();
+      for ( auto next = index + 1; next <= this->numberIncidentEnergies(); ++next ) {
+
+        auto nextlocator = this->relativeAngularDistributionLocator( next );
+        if ( nextlocator > locator ) {
+
+          right = std::next( this->begin(), nextlocator - 1 );
+          break;
+        }
+      }
```

Comment:
This is another example of why you should not store multiple pieces of information in a single field. When the locator is zero it means the distribution is isotropic so it is possible to have the following locators:
```
1   24    0     0   48
``` 
The sub-block that starts at 24 for the second incident energy stops at position 48 - 1, which is just before the data for the 4th incident energy value. I therefore need to go over the different locators and find the first one that is either not zero or larger than the current locator to know where my sub-block stops.

Technically speaking this should NEVER happen since isotropic distributions only happen for low incident energies after which they become anisotropic. However, if it would happen, it would cause an error in ACEtk hence the code protecting itself against a very unlikely situation.